Sources/SwiftGRPCNIO/GRPCError.swift

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2019, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Foundation
+
+public enum GRPCError: Error, Equatable {
```

Comment:
Thank you, and well done!

Idea: Should we have the server-only errors in a separate enum, so that client-side users don't need to care for server-only errors, and vice versa?

Reply:
I was thinking about splitting them too; is having all three actually an issue?

Reply:
Might make it slightly harder to `switch` over them, but it would probably still make sense to split them, to avoid duplication for common errors.

Reply:
For the time being I just renamed it to `GRPCServerError`. The common cases can be extracted when I resolve conflicts with the client PR.

At the moment I think the only place we switch over them is in the user-provided error delegate. We _could_ have different methods for the different enum types:

```
func observe(error: GRPCServerError)
func observe(error: GRPCCommonError)
// fallback to this:
func observe(error: Error)
```

We can default to just calling the `Error` observer for the server and common cases.

Reply:
I like the idea of having different methods; agreed that this should be the topic of a later PR, though.